core/src/main/scala/com/mesosphere/usi/core/ResourceUtil.scala

```diff
@@ -0,0 +1,218 @@
+package com.mesosphere.usi.core
+
+import com.github.ghik.silencer.silent
+import com.typesafe.scalalogging.StrictLogging
+import org.apache.mesos.v1.Protos.Resource.DiskInfo.Source
+import org.apache.mesos.v1.Protos.Resource.{DiskInfo, ReservationInfo}
+import org.apache.mesos.v1.{Protos => Mesos}
+import scala.collection.JavaConverters._
+import scala.util.control.NonFatal
+
+object ResourceUtil extends StrictLogging {
```

Comment:
note for reviewer: this code seems to be basically copy&paste from Marathon. I'll follow up with migrating the tests as well

Reply:
I didn't see this, I already migrated the tests.